src/lib/syncPagePath.ts

```diff
@@ -0,0 +1,418 @@
+import { categoryUriRegexPattern, parentUriRegexPattern } from './types.js';
+
+/**
+ * Commands that use this file for syncing Markdown via APIv2.
+ *
+ * Note that the `changelogs` command is not included here
+ * because it is backed by APIv1.
+ */
+export type CommandsThatSyncMarkdown = DocsUploadCommand;
+
+type PageRepresentation = GuidesRequestRepresentation;
+
+interface FailedPushResult {
+  error: APIv2Error | Error;
+  filePath: string;
+  result: 'failed';
+  slug: string;
+}
+
+type PushResult =
+  | FailedPushResult
+  | {
+      filePath: string;
+      response: PageRepresentation | null;
+      result: 'created' | 'skipped' | 'updated';
+      slug: string;
+    };
+
+/**
+ * Reads the contents of the specified Markdown or HTML file
+ * and creates/updates the corresponding page in ReadMe
+ */
+async function pushPage(
+  this: CommandsThatSyncMarkdown,
+  /** the file data */
+  fileData: PageMetadata,
+): Promise<PushResult> {
+  const { key, 'dry-run': dryRun, version } = this.flags;
+  const { content, filePath, slug } = fileData;
+  const data = fileData.data;
+  let route = `/${this.route}`;
+  if (version) {
+    route = `/versions/${version}${route}`;
+  }
+
+  const headers = new Headers({ authorization: `Bearer ${key}`, 'Content-Type': 'application/json' });
+
+  if (!Object.keys(data).length) {
+    this.debug(`No frontmatter attributes found for ${filePath}, not syncing`);
+    return { response: null, filePath, result: 'skipped', slug };
+  }
+
+  const payload: PageRepresentation = {
+    ...data,
+    content: {
+      ...(typeof data.content === 'object' ? data.content : {}),
+      body: content,
+    },
+    slug,
+  };
+
+  try {
+    // normalize the category uri
+    if (payload.category?.uri) {
+      const regex = new RegExp(categoryUriRegexPattern);
+      if (!regex.test(payload.category.uri)) {
+        let uri = payload.category.uri;
+        this.debug(`normalizing category uri ${uri} for ${filePath}`);
+        // remove leading and trailing slashes
+        uri = uri.replace(/^\/|\/$/g, '');
+        payload.category.uri = `/versions/${version}/categories/${this.route}/${uri}`;
+      }
+    }
+
+    // normalize the parent uri
+    if (payload.parent?.uri) {
+      const regex = new RegExp(parentUriRegexPattern);
+      if (!regex.test(payload.parent.uri)) {
+        let uri = payload.parent.uri;
+        this.debug(`normalizing parent uri ${uri} for ${filePath}`);
+        // remove leading and trailing slashes
+        uri = uri.replace(/^\/|\/$/g, '');
+        payload.parent.uri = `/versions/${version}/${this.route}/${uri}`;
+      }
+    }
+
+    const createPage = (): Promise<PushResult> | PushResult => {
+      if (dryRun) {
+        return { filePath, result: 'created', response: null, slug };
+      }
+
+      return this.readmeAPIFetch(
+        route,
+        { method: 'POST', headers, body: JSON.stringify(payload) },
+        { filePath, fileType: 'path' },
```

Comment:
What's this third argument for?

Reply:
this:

https://github.com/readmeio/rdme/blob/9f84783d19556032ec9e2b911be20630a657b408/src/lib/readmeAPIFetch.ts#L264-L278